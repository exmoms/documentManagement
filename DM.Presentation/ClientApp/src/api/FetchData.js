/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
