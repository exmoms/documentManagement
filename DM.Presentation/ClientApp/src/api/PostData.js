/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
