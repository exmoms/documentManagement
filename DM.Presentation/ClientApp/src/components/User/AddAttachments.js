/**
 * @file Contains AddAttachments component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import React, { Component } from "react";

import CREATE_DOC_STYLES from "./Styles";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import CloudUploadIcon from "@material-ui/icons/CloudUpload";
import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContentText from "@material-ui/core/DialogContentText";
import { withTranslation } from "react-i18next";
const styles = CREATE_DOC_STYLES;
/**
 * Two required props:
 *  - updater: handle changes of attachments and update it for parent component
 *  - compoundModelsInfo: contains the info of the required attachments
 */
export class AddAttachments extends Component {
  inputRefs = new Set();

  constructor(props) {
    super(props);

    this.state = {
      /**
       * list of objects
       * {
       *   compoundModelId: [int],
       *   attachmentName: [string]
       *   attachedFile: [file/pathOnServer]
       *   isRequiered: [bool]
       * }
       */
      isCompleted: false,
      attachments: this.props.compoundModelsInfo,
      show: false,
      index: -1,
    };
    this.updateParentAttachments = this.props.updater;
  }
  t = this.props["t"];

  componentDidUpdate(prevProps) {
    if (this.props.compoundModelsInfo !== prevProps.compoundModelsInfo) {
      this.inputRefs = new Set();
      this.setState({ attachments: this.props.compoundModelsInfo });
    }
  }

  addAttachmentcOnClickHandler = (compoundModel) => {
    /**
     * @todo handle how to add an attachment
     */

    var temp = [...this.inputRefs];
    var index = temp.findIndex((item) => {
      if (item === null) {
        return false;
      } else {
        return parseInt(item.id) === compoundModel.compoundModelId;
      }
    });
    if (index !== -1) {
      var fileUploader = temp[index];
      fileUploader.click();
    }
  };

  onChangeFile = (event) => {
    event.stopPropagation();
    event.preventDefault();
    var file = event.target.files[0];
    let id_ = parseInt(event.target.id);

    if (file !== undefined && file !== null) {
      var index_ = -1;
      index_ = this.state.attachments.findIndex((item) => {
        return item.compoundModelId === id_;
      });

      if (index_ === -1) {
        if (process.env.NODE_ENV === "development") {
          console.log("[ERROR] Can't find the event.target.id:", id_);
        }
      } else {
        /* Check if the file had never been add, no duplicated attachments */
        let valid = this.state.attachments.every((item, idx) => {
          let valid_ = true;
          if (idx !== index_) {
            if (item.attachedFile !== undefined) {
              valid_ = item.attachedFile.name !== file.name;
            }
          }
          return valid_;
        });

        if (valid) {
          let temp = this.state.attachments;
          temp[index_].attachedFile = file;
          let validateAttachments = temp.every((item) => {
            let requiredIsAttached =
              item.isRequiered && item.attachedFile !== undefined;
            let unrequiredIsNotAttachedOrAttached = !item.isRequiered;

            return requiredIsAttached || unrequiredIsNotAttachedOrAttached;
          });

          /// we want to upload latter
          this.setState({
            attachments: temp,
            isCompleted: validateAttachments,
          });
          /**
           * @todo add functionality to delete the attched file
           * maybe user after his selection changes his mind
           */
          document.getElementById(
            "table-cell-aggregated-doc" + temp[index_].compoundModelId
          ).innerHTML = file.name;
          this.updateParentAttachments(temp, validateAttachments);
        } else {
          /* case of the attached file is duplicated */
          /// reset input element to trige onChange
          event.target.value = null;

          let temp = this.state.attachments;
          temp[index_].attachedFile = undefined;

          this.setState({
            index: index_,
            show: true,
            isCompleted: false,
            attachments: temp,
          });
        }
      }
    }
  };

  renderCompoundModelsTable = () => {
    const { attachments } = this.state;
    const { classes } = this.props;
    /**
     * @todo This table should render one time and one more for each time we change/switch
     * to another model
     */
    if (this.props.compoundModelsInfo.length !== 0) {
      return (
        <TableContainer component={Paper}>
          <Table className={classes.table} aria-label="simple table">
            <TableHead>
              <TableRow>
                <TableCell>{this.t("namelabel")}</TableCell>
                <TableCell>{this.t("actions")}</TableCell>
                <TableCell>{this.t("attachedfilename")}</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {attachments.map((attachment, index) => {
                // Once the fileUploader must be differnent for each upload button
                // let fileUploaderName = `fileUploader${compoundModel.compoundModelId}`;
                return (
                  <TableRow key={attachment.compoundModelId}>
                    <TableCell>
                      {attachment.isRequiered
                        ? attachment.attachmentName + " *"
                        : attachment.attachmentName}
                    </TableCell>
                    <TableCell>
                      <Button
                        id={"addAttachment" + attachment.compoundModelId}
                        onClick={(e) => {
                          console.log(
                            "call the click event on an attachment with id " +
                              attachment.compoundModelId
                          );
                          this.addAttachmentcOnClickHandler(attachment);
                        }}
                        variant="contained"
                        color="primary"
                        size="small"
                        className={classes.buttonTable}
                        startIcon={<CloudUploadIcon />}
                      >
                        <input
                          type="file"
                          className={
                            "onChangeFile" + attachment.compoundModelId
                          }
                          id={attachment.compoundModelId}
                          required={attachment.isRequiered}
                          ref={(ref) => {
                            this.inputRefs.add(ref);
                          }}
                          style={{ display: "none" }}
                          onChange={this.onChangeFile}
                        />
                        {this.t("button_attach")}
                      </Button>
                    </TableCell>
                    <TableCell
                      id={
                        "table-cell-aggregated-doc" + attachment.compoundModelId
                      }
                    >
                      {this.state.attachments[index].attachedFile
                        ? this.state.attachments[index].attachedFile.name
                        : "..."}
                    </TableCell>
                  </TableRow>
                );
              })}
            </TableBody>
          </Table>
        </TableContainer>
      );
    }
  };

  toggleShow = (show) => {
    this.setState({
      show: show,
      index: -1,
    });
  };

  renderDuplicatedAttachmentsDialog = () => {
    if (this.state.index !== -1) {
      return (
        <div>
          <Dialog
            position="centerCenter"
            open={this.state.show}
            onClose={() => this.toggleShow(false)}
            aria-labelledby="form-dialog-title"
          >
            <DialogTitle id="form-dialog-title">
              {this.t("dialog_duplicatedattachment")}
            </DialogTitle>
            <DialogContent>
              <DialogContentText>
                {this.t("dialog_selectedbefore")}
              </DialogContentText>
            </DialogContent>
            <DialogActions>
              <Button
                id={"addAttachmentcOnClickHandlerButton" + this.state.index}
                onClick={
                  /* re-open file browser to select another file */
                  (e) => {
                    console.log(
                      "call the click event on an attachment with id " +
                        this.state.attachments[this.state.index].compoundModelId
                    );
                    this.addAttachmentcOnClickHandler(
                      this.state.attachments[this.state.index]
                    );
                    this.toggleShow(false);
                  }
                }
                color="primary"
              >
                {this.t("button_addfile")}
              </Button>
              <Button
                onClick={() => this.toggleShow(false)}
                id="toggleShowButton"
                color="primary"
              >
                {this.t("button_cancel")}
              </Button>
            </DialogActions>
          </Dialog>
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        <div>{this.renderCompoundModelsTable()}</div>
        <div>{this.renderDuplicatedAttachmentsDialog()}</div>
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(AddAttachments));
