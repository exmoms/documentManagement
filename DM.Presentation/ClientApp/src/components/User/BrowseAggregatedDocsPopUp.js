/**
 * @file Contains BrowseAggregatedDocsPopUp component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import React, { Component } from "react";
import DocumentTable from "./DocumentTable";
import SearchDocuments from "./SearchDocuments";
import CREATE_DOC_STYLES from "./Styles";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContentText from "@material-ui/core/DialogContentText";

import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";

import { withTranslation } from "react-i18next";

const styles = CREATE_DOC_STYLES;

class BrowseAggregatedDocsPopUp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      obj: {
        id: -1,
        name: "...",
        latest_version: -1,
      },
      data: [],
      columns: [],
    };
    this.updateParentState = this.props.action;
  }
  t = this.props["t"];

  getDocumentList = (list, columns) => {
    this.setState({
      data: list,
      columns: columns,
    });
  };

  toggleShow = (show) => {
    this.updateParentState("show", show);
  };

  handleSelect = (value) => {
    if (value !== null) {
      this.updateParentState("addDoc", {
        modelId: this.props.modelId,
        selectedDoc: value,
      });

      this.toggleShow(false);
    }
  };

  /**
   * @todo Could we reduce rendering this component, once it renders two times
   * each time we update state
   */
  render() {
    const { classes } = this.props;
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.toggleShow(false)}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">
            {this.t("addaggregateddoc")}
          </DialogTitle>
          <DialogContent>
            <DialogContentText>
              {this.t("dialog_selectaggregateddocument")} <strong>Hint:</strong>{" "}
              {this.t("dialog_previewchooseddocument")}
            </DialogContentText>
            <TableContainer component={Paper}>
              <Table className={classes.table} aria-label="simple table">
                <TableBody>
                  <TableRow key="search-form">
                    <TableCell>
                      <SearchDocuments
                        getDocumentList={this.getDocumentList}
                        action="update-document"
                        modelId={this.props.modelId}
                      />
                      <DocumentTable
                        data={this.state.data}
                        columns={this.state.columns}
                        action="update-document"
                        handleSelect={this.handleSelect}
                      />
                    </TableCell>
                  </TableRow>
                </TableBody>
              </Table>
            </TableContainer>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.toggleShow(false)}
              color="primary"
            >
              {this.t("button_cancel")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(BrowseAggregatedDocsPopUp));
