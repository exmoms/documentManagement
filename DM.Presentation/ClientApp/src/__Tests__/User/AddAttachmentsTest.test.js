import React from "react";
import ReactDom from "react-dom";
import AddAttachments from "../../components/User/AddAttachments";
import BrowseAggregatedDocsPopUp from "../../components/User/BrowseAggregatedDocsPopUp";
import { getAvailableAggregatedDocs } from "../../api/FetchData";
import { cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import Dialog from "@material-ui/core/Dialog";
import DialogContent from "@material-ui/core/DialogContent";
import Autocomplete from "@material-ui/lab/Autocomplete";
import Input from "@material-ui/core/Input";
import Button from "@material-ui/core/Button";
afterEach(cleanup);
describe("AddAttachments", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    let compoundModelsInfo = [
      { compoundModelId: 1, attachedFile: { id: 2, name: "AttachedDoc2" } },
    ];
    const updater = jest.fn();
    const div = document.createElement("div");
    ReactDom.render(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />,
      div
    );
  });

  test("AddAttachments matches snapshot", () => {
    let compoundModelsInfo = [
      { compoundModelId: 1, attachedFile: { id: 2, name: "AttachedDoc2" } },
    ];
    const updater = jest.fn();
    const component = renderer.create(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("AddAttachments addAttachmentcOnClickHandler functionality", () => {
    //TBD
    let compoundModelsInfo = [
      {
        compoundModelId: 1,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
      {
        compoundModelId: 2,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
    ];
    const updater = jest.fn();
    const clickEvent = jest.fn();
    let wrapper = mount(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />,
      { attachTo: document.body }
    );
    // document.getElementById("addAttachment1").addEventListener('click', clickEvent)
    // const mockClick = jest.spyOn(document.getElementById("addAttachment1"), 'click')
    const mockConsole = jest.spyOn(console, "log");
    //click addAttachmentcOnClickHandler
    act(() => {
      wrapper
        .find("#addAttachment1")
        .at(0)
        .props()
        .onClick({ target: { value: "" } });
    });
    wrapper.update();
    expect(mockConsole).toHaveBeenCalledTimes(2);
    expect(mockConsole.mock.calls[0][0]).toBe(
      "call the click event on an attachment with id 1"
    );
    expect(mockConsole.mock.calls[1][0]).toBe(
      "call the click event on an attachment with id 1"
    );
    // expect(mockClick).toHaveBeenCalledTimes(1);
  });

  test("AddAttachments onChangeFile functionality and is valid", async () => {
    let compoundModelsInfo = [
      {
        compoundModelId: 1,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
    ];
    const updater = jest.fn();
    const stopPropagationMock = jest.fn();
    const preventDefaultMock = jest.fn();
    let wrapper = mount(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />
    );
    //click addAttachmentcOnClickHandler
    expect(wrapper.find(".onChangeFile1").length).toBe(1);
    act(() => {
      wrapper
        .find(".onChangeFile1")
        .at(0)
        .props()
        .onChange({
          target: { id: 1, files: [{ name: "file1" }] },
          stopPropagation: stopPropagationMock,
          preventDefault: preventDefaultMock,
        });
    });
    wrapper.update();
    expect(stopPropagationMock).toHaveBeenCalledTimes(1);
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(
      document.getElementById("table-cell-aggregated-doc1").innerHTML
    ).toBe("file1");
    expect(updater).toHaveBeenCalledTimes(1);
    //first argument for first call of the mock
    let temp = compoundModelsInfo;
    let file = { name: "file1" };
    let index_ = -1;
    index_ = compoundModelsInfo.findIndex((item) => {
      return item.compoundModelId === 1;
    });
    temp[index_].attachedFile = file;
    expect(updater.mock.calls[0][0]).toBe(temp);
    expect(updater.mock.calls[0][1]).toBeTruthy();
  });

  test("AddAttachments onChangeFile functionality and is not valid", async () => {
    let compoundModelsInfo = [
      {
        compoundModelId: 1,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
      {
        compoundModelId: 2,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
    ];
    const updater = jest.fn();
    const stopPropagationMock = jest.fn();
    const preventDefaultMock = jest.fn();
    let wrapper = mount(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />
    );
    //click addAttachmentcOnClickHandler
    expect(wrapper.find(".onChangeFile1").length).toBe(1);
    act(() => {
      wrapper
        .find(".onChangeFile1")
        .at(0)
        .props()
        .onChange({
          target: { id: 1, files: [{ name: "AttachedDoc2" }] },
          stopPropagation: stopPropagationMock,
          preventDefault: preventDefaultMock,
        });
    });
    wrapper.update();
    expect(stopPropagationMock).toHaveBeenCalledTimes(1);
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(
      wrapper.find("#table-cell-aggregated-doc1").innerHTML
    ).toBeUndefined();
    expect(wrapper.find(Dialog).props().open).toBeTruthy();
  });

  test("AddAttachments renderDuplicatedAttachmentsDialog functionality , toggleShow functionality", async () => {
    let compoundModelsInfo = [
      {
        compoundModelId: 1,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
      {
        compoundModelId: 2,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
    ];
    const updater = jest.fn();
    const stopPropagationMock = jest.fn();
    const preventDefaultMock = jest.fn();
    let wrapper = mount(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />
    );
    //click addAttachmentcOnClickHandler to open the dialog
    act(() => {
      wrapper
        .find(".onChangeFile1")
        .at(0)
        .props()
        .onChange({
          target: { id: 1, files: [{ name: "AttachedDoc2" }] },
          stopPropagation: stopPropagationMock,
          preventDefault: preventDefaultMock,
        });
    });
    wrapper.update();
    expect(stopPropagationMock).toHaveBeenCalledTimes(1);
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(
      wrapper.find("#table-cell-aggregated-doc1").innerHTML
    ).toBeUndefined();
    expect(wrapper.find(Dialog).props().open).toBeTruthy();
    //click onClose for dialog
    act(() => {
      wrapper
        .find(Dialog)
        .props()
        .onClose({ target: { value: "" } });
    });
    wrapper.update();
    expect(wrapper.find(Dialog).innerHTML).toBeUndefined();
  });

  test("AddAttachments renderDuplicatedAttachmentsDialog functionality , addAttachmentcOnClickHandler functionality", async () => {
    let compoundModelsInfo = [
      {
        compoundModelId: 1,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
      {
        compoundModelId: 2,
        attachedFile: { id: 2, name: "AttachedDoc2" },
        isRequiered: true,
      },
    ];
    const updater = jest.fn();
    const stopPropagationMock = jest.fn();
    const preventDefaultMock = jest.fn();
    const mockConsole = jest.spyOn(console, "log");
    let wrapper = mount(
      <AddAttachments
        updater={updater}
        compoundModelsInfo={compoundModelsInfo}
      />
    );
    //click addAttachmentcOnClickHandler to open the dialog
    act(() => {
      wrapper
        .find(".onChangeFile1")
        .at(0)
        .props()
        .onChange({
          target: { id: 1, files: [{ name: "AttachedDoc2" }] },
          stopPropagation: stopPropagationMock,
          preventDefault: preventDefaultMock,
        });
    });
    wrapper.update();
    expect(stopPropagationMock).toHaveBeenCalledTimes(1);
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(
      wrapper.find("#table-cell-aggregated-doc1").innerHTML
    ).toBeUndefined();
    expect(wrapper.find(Dialog).props().open).toBeTruthy();
    //click onClose for dialog
    act(() => {
      wrapper
        .find("#addAttachmentcOnClickHandlerButton0")
        .at(0)
        .props()
        .onClick({ target: { value: "" } });
    });
    wrapper.update();
    expect(mockConsole).toHaveBeenCalledTimes(1);
    expect(mockConsole.mock.calls[0][0]).toBe(
      "call the click event on an attachment with id 1"
    );
    // toggleShow
    expect(wrapper.find(Dialog).innerHTML).toBeUndefined();
  });
});
