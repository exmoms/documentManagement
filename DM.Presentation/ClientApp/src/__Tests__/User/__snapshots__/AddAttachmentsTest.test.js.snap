// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAttachments AddAttachments matches snapshot 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root withI18nextTranslation(AddAttachments)-table-6"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              namelabel
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              actions
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              attachedfilename
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained withI18nextTranslation(AddAttachments)-buttonTable-11 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                id="addAttachment1"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiButton-startIcon MuiButton-iconSizeSmall"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                      />
                    </svg>
                  </span>
                  <input
                    className="onChangeFile1"
                    id={1}
                    onChange={[Function]}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="file"
                  />
                  button_attach
                </span>
              </button>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="table-cell-aggregated-doc1"
            >
              AttachedDoc2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div />
</div>
`;
